lib/containers/container_images.pm

```diff
@@ -208,11 +209,11 @@ sub test_zypper_on_container {
         record_info('bsc#1192941', 'bsc#1192941 - zypper-docker entrypoint confuses program arguments');
     }
     validate_script_output("$runtime run -i --entrypoint '' $image zypper lr -s", sub { m/.*Alias.*Name.*Enabled.*GPG.*Refresh.*Service/ }, timeout => 180);
-    script_retry("$runtime run -i --name 'refreshed' --entrypoint '' $image zypper -nv ref", timeout => 300, retry => 3, delay => 60);
+    assert_script_run("$runtime run -t -d --name 'refreshed' --entrypoint '' $image bash", timeout => 300);
+    script_retry("$runtime exec refreshed zypper -nv ref", timeout => 600, retry => 3, delay => 60);
     assert_script_run("$runtime commit refreshed refreshed-image", timeout => 120);
     assert_script_run("$runtime rm -f refreshed");
-    script_retry("$runtime run -i --name 'refreshed-image' --rm --entrypoint '' $image zypper -nv ref", timeout => 300, retry => 3, delay => 60);
-    record_info "The End", "zypper test completed";
+    script_retry("$runtime run -i --rm --entrypoint '' refreshed-image zypper -nv ref", timeout => 300, retry => 3, delay => 60);
```

Comment:
Why don't we use `$runtime exec` over here as well?

Reply:
What I understand from this flow is:
1) Run zypper ref in container and commit the state of the container to a new local image `refreshed-image`.
2) Run another container from that committed local image

For 1) we CAN'T use script_retry because we force the `--name` in the command. If the command fails or timesout, the container might have been created and the next retry will fail. So, that's why I opted for creating the container and leave it running, and then retry ONLY the `zypper ref` command. Then, commit.

For 2) we dont' need to provide a `--name` to the container, because we create it from a local image and don't care about the container name. Docker/podman will create a random name. So, we can use `script_retry` directly there.